cli/flox/src/commands/environment.rs

```diff
@@ -423,18 +424,47 @@ impl Activate {
         let now_active =
             UninitializedEnvironment::from_concrete_environment(&concrete_environment)?;
 
-        let mut environment = concrete_environment.into_dyn_environment();
+        let environment = concrete_environment.dyn_environment_ref_mut();
 
         // Don't spin in bashrcs and similar contexts
-        let activation_path = if !stdout().is_tty() && self.run_args.is_empty() {
-            environment.activation_path(&flox)?
+        let activation_path_result = if !stdout().is_tty() && self.run_args.is_empty() {
+            environment.activation_path(&flox)
         } else {
             Dialog {
                 message: &format!("Getting ready to use environment {now_active}..."),
                 help_message: None,
                 typed: Spinner::new(|| environment.activation_path(&flox)),
             }
-            .spin()?
+            .spin()
+        };
+
+        let activation_path = match activation_path_result {
+            Err(
+                err @ EnvironmentError2::Core(CoreEnvironmentError::LockedManifest(
+                    LockedManifestError::BuildEnv(CallPkgDbError::PkgDbError(PkgDbError {
+                        exit_code: 123,
+                        ..
+                    })),
+                )),
+            ) => {
+                let mut message = formatdoc! {"
+                    ❌ This environment is not yet compatible with your system ({system}).
+
+                    {err}",
```

Comment:
blocking: since you're catching specifically 123 there's no need to print the error now, right?